src/forms/labels/field-set-legend.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import classnames from 'classnames'
+import { convertNameToLabel } from '../helpers'
+
+/**
+ *
+ * A legend representing a caption for the content of its parent field set element
+ *
+ * This component must be used as a direct child and the only legend of the <fieldset> element that groups related controls
+ * 
+ *
+ * The text of the legend is set using the following rules:
+ * - If the `label` prop is set to `false`, the legend is hidden visually
+ * - Else If the `label` prop is set to a string, the label will display that text
+ * - Otherwise, the label will be set using the `name` prop.
+ *
+ * 
+ * @name FieldSetLegend
+ * @type Function
+ * @param {String} name - The name of the associated group
+ * @param {String} [hint] - A usage hint for the associated input
+ * @param {String|Boolean} [label] - Custom text for the legend
+ * @param {Boolean} [required] - A boolean value to indicate whether the field is required
```

Comment:
This should have a default. You can communicate that with `=` within the brackets